### PR TITLE
Doc Page for Customer-Managed Deployment Agents

### PR DESCRIPTION
**Pull request by arunkumar611**
Will be re-recording and updating new mp4s and images as the current version is too small. 

Fix https://github.com/pulumi/pulumi-service/issues/17378

### CODE REVIEW

File: themes/default/content/docs/pulumi-cloud/deployments/customer-managed-deployments.md [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+- **Multiple pools**: You can set up multiple deployment agent pools, assign stacks to specific pools, and scale agents dynamically to increase your deployment concurrency. Customers can have up to 150 concurrent deployments
+- **Meet Compliance**: You can configure the agents with the credentials needed to manage your infrastructure. This way your cloud provider credentials never leave your private network
+
+<sup>1</sup> *Currently Linux and MacOS are supported*
+
+{{% notes "info" %}}
+Customer-managed deployment agents are available to business-critical customers. Contact sales if you are interested and want to enable customer-managed deployments.
+{{% /notes %}}
+
+Before you begin, ensure you have installed [Pulumi Github App](/docs/using-pulumi/continuous-delivery/github-app/) and update the [source control settings](/docs/pulumi-cloud/deployments/get-started/#:~:text=Fill%20out%20the%20Source%20control%20settings). [Docker](https://docs.docker.com/engine/) is a requirement for the agent.
```
Comment (glena) [inline themes/default/content/docs/pulumi-cloud/deployments/customer-managed-deployments.md:25]:
```suggestion
Before you begin, ensure you have installed [Pulumi Github App](/docs/using-pulumi/continuous-delivery/github-app/) and update the [source control settings](/docs/pulumi-cloud/deployments/get-started/#:~:text=Fill%20out%20the%20Source%20control%20settings). [Docker](https://docs.docker.com/engine/) is a requirement for using  deployments.
```

File: themes/default/content/docs/pulumi-cloud/deployments/customer-managed-deployments.md [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+
+## How to install and configure the agent
+
+Go to **Deployment runners** under settings and click on Add a new pool. Provide a name and description. You will get a new access token for the pool. Ensure to copy and save the token in a secure location.
+
+{{< video title="Install the agents" src="../customer-managed-deployment-agents-assets/install-the-agents.mp4" autoplay="true" loop="true" >}}
+
+{{< notes type="warning" >}}
+If you run into issues, ensure docker is running before you configure and start the agents. ensure you restart or refresh the shell session (example: `source ~/.zshrc`).
```
Comment (glena) [inline themes/default/content/docs/pulumi-cloud/deployments/customer-managed-deployments.md:34]: maybe link to https://docs.docker.com/engine/install/

File: themes/default/content/docs/pulumi-cloud/deployments/customer-managed-deployments.md [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+{{< /notes >}}
+
+{{< video title="Configure the agents" src="../customer-managed-deployment-agents-assets/configure-the-agents.mp4" autoplay="true" loop="true" >}}
+
+Now your first agent should be up and running. You can verify the status of the agent by refreshing the page. Additionally, you can use the same token to set up multiple agents to increase concurrency of your deployments and they will be assigned to the same pool.
```
Comment (glena) [inline themes/default/content/docs/pulumi-cloud/deployments/customer-managed-deployments.md:39]: should the doc include the code to install, configure and run? (same we added in the console)

File: themes/default/content/docs/pulumi-cloud/deployments/customer-managed-deployments.md [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+   ```bash
+   VARIABLE=value customer-managed-deployment-agent run
+   ```
+
+   You also need to update the `pulumi-deployment-agent.yaml` configuration file by setting `env_forward_allowlist`. The configuration file can be found in the directory where the agent is extracted. `env_forward_allowlist` expects an array of strings. Example:
+
+    ```yaml
+    token: pul-d2d2….
+    version: v0.0.5
+    env_forward_allowlist:
```
Comment (glena) [inline themes/default/content/docs/pulumi-cloud/deployments/customer-managed-deployments.md:65]: maybe we should mention this is case sensitive (if you allowlist key_one, you cannot use KEY_ONE when setting the variable)

Comment (arunkumar611) [reply themes/default/content/docs/pulumi-cloud/deployments/customer-managed-deployments.md:65]: If its case sensitive I can use both key_one and KEY_ONE?

File: themes/default/content/docs/pulumi-cloud/deployments/customer-managed-deployments.md (hunk 3 above)
Comment (arunkumar611) [reply themes/default/content/docs/pulumi-cloud/deployments/customer-managed-deployments.md:39]: I think this is fine as our steps within the deployments runner page are pretty good..

File: themes/default/content/docs/pulumi-cloud/deployments/customer-managed-deployments.md (hunk 2 above)
Comment (arunkumar611) [reply themes/default/content/docs/pulumi-cloud/deployments/customer-managed-deployments.md:34]: updating this in the new commit

**Pull request closed without merging**